packages/queries/src/query.ts

```
@@ -343,172 +195,97 @@ export class QueryBuilder<
      *   });
      * ```
      */
-    columns<TKeys extends Array<Column<DB, TTable>>>(...keys: TKeys) {
+    columns<TKeys extends Array<Column<DB, TTable>>>(
+        ...keys: TKeys
+    ): QueryBuilder<DB, TTable, TPick<TResultSchema, TKeys>> {
         type SelectFromKeys = { [k in TKeys[number]]: true };
 
         const select = keys.reduce((acc, key) => {
             return { ...acc, [key]: true };
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)